### PR TITLE
Unit test for hashing module

### PR DESCRIPTION
**Pull request by NoCtrlZ**
I wrote unit test for hashing module.
We need to change the HashSize or base58.Decode function.

### CODE REVIEW

File: plugins/qnode/hashing/hash_test.go [hunk 1]
```diff
@@ -3,10 +3,79 @@ package hashing
...
+func (s *SampleSource) Int63() int64 {
+	return s.seed
+}
+
+func (s *SampleSource) Seed(seed int64) {
+	fmt.Println(s)
+}
+
+func TestHashValueFromString(t *testing.T) {
+	var text = "string text"
```
Comment (lunfardo314) [inline plugins/qnode/hashing/hash_test.go:22]: the text must be base58 encoded string, which when decoded gives 32 bytes long hash.
Best way is something like this:
hash1 := HashStrings("test string")
encodedHash  := base58.Encode(hash1)
hash2 := HashValueFromString(hash1)
 if !hash1.Equal(hash2){
     t.Errorf("failed")
}

File: plugins/qnode/hashing/hash_test.go [hunk 2]
```diff
@@ -3,10 +3,79 @@ package hashing
...
+	fmt.Printf("%x len = %d bytes\n", h, len(h))
+}
+
+func TestHashDataSha3(t *testing.T) {
+	var bytes = []byte{0, 1, 2, 3}
+	h := HashDataSha3(bytes)
+	fmt.Printf("%x len = %d bytes\n", h, len(h))
+}
+
 func TestHashStrings(t *testing.T) {
```
Comment (lunfardo314) [inline plugins/qnode/hashing/hash_test.go:48]: it is better to rewrite the test completely

File: plugins/qnode/hashing/hash_test.go [hunk 3]
```diff
@@ -3,10 +3,79 @@ package hashing
...
+	var src = &SampleSource{
+		seed: 1,
+	}
+	var rnd = rand.New(src)
+	h := RandomHash(rnd)
+	fmt.Printf("%x len = %d bytes\n", h[:], len(h[:]))
+}
+
+func TestHashInList(t *testing.T) {
+	var seed1 = "alice"
```
Comment (lunfardo314) [inline plugins/qnode/hashing/hash_test.go:64]: see comment above. HashValueFromString creates 32 byte long HashValue bu decoding it woth base58.Decode. So, the argument cannot be arbitrary string

File: plugins/qnode/hashing/hash.go [hunk 4]
```diff
@@ -74,6 +74,7 @@ func (h *HashValue) UnmarshalJSON(buf []byte) error {
 }
 
 func HashValueFromString(s string) (*HashValue, error) {
+	// should be same length as HashSize
```
Comment (lunfardo314) [inline plugins/qnode/hashing/hash.go:77]: // should be base58 encoded string which when decoded must give 32 bytes long hash value

File: plugins/qnode/hashing/hash_test.go (hunk 1 above)
Comment (NoCtrlZ) [reply plugins/qnode/hashing/hash_test.go:22]: Thank you.
It's cool!
I am gonna fix as you mentioned.

**Pull request merged**